src/async-syntax-highlighter.js

```diff
@@ -15,41 +17,84 @@ export default (options) => {
       return ReactAsyncHighlighter.loadAstGenerator();
     }
 
-    static registerLanguage = (_, language) => {
-      if (!supportsLanguageRegistering) {
+    static async loadLanguage(language) {
+      const languageLoader = languageLoaders[language];
+      if (typeof languageLoader === 'function') {
+        return languageLoader(ReactAsyncHighlighter.registerLanguage);
+      } else {
+        throw `Language ${language} not supported`
+      }
+    }
+    
+    static isRegistered = (language) => {
+      if(!registerLanguage) {
```

Comment:
should we return true here? what would be the case where we are using async but don't have register language, i would lean towards throwing